instrumentation_sql.go

```diff
@@ -227,11 +227,38 @@ func (conn *wrappedSQLConn) ExecContext(ctx context.Context, query string, args
 }
 
 func (conn *wrappedSQLConn) CheckNamedValue(d *driver.NamedValue) error {
+	defer func() {
+		if err := recover(); err != nil {
+			sensor.logger.Debug("Swallowing error: ", err)
+		}
+	}()
+
 	if c, ok := conn.Conn.(driver.NamedValueChecker); ok {
 		return c.CheckNamedValue(d)
 	}
 
-	return nil
+	q := "create table instanatemp (a varchar(1))"
+
+	stmt, err := conn.Prepare(q)
+
+	if err != nil {
+		sensor.logger.Debug("Database does not support temporary statement: ", q)
+	}
+
+	defer stmt.Close()
+
+	if s, ok := stmt.(driver.NamedValueChecker); ok {
+		return s.CheckNamedValue(d)
+	}
+
+	d.Value, err = driver.DefaultParameterConverter.ConvertValue(d.Value)
```

Comment:
If both `driver.Conn` and `driver.Stmt` don't implement `CheckNamedValue` we fallback to Go's standard lib default validation and conversion method.

Reply:
The most important part of the fix is to assure that a converter is evaluated in order to make sure that a value type is valid within the query. eg: providing `int32` to the `pq` driver is invalid and should be converted to `int64`, and so on.